streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1193,6 +1189,18 @@ public void updateTaskEndMetadata(final TopicPartition topicPartition, final Lon
         }
     }
 
+    /**
+     * Checks for added or removed NamedTopologies that correspond to any assigned tasks, and creates/freezes them if so
+     */
+    void handleTopologyUpdates() {
+        tasks.maybeCreateTasksFromNewTopologies();
+        for (final Task task : activeTaskIterable()) {
+            if (topologyMetadata.namedTopologiesView().contains(task.id().namedTopology())) {
+                task.freezeProcessing();
```

Comment:
Not sure I understand this logic.. could you explain a bit? Also there's no unfrozen function so once a task is frozen it would stick with that state forever?

Reply:
This is another stupid bug...it should be freezing them if they _don't _ appear in the current topology. Tasks from removed named topologies are frozen to prevent them from processing any more records, so yes, it is meant to be permanent.

Reply:
Just another thought: if this is just for the very short temporary phase between when the topology is removed to when the rebalance is finally triggered to remove the tasks (which should usually be in the next poll), could we just call `task.suspend` instead of adding the new `freeze` logic? When we finally close the tasks we can still transit from suspended to closed?

Reply:
Well long story short I was trying to avoid mucking around in the task state management (which has historically been the source of many critical bugs) -- also we want to remove the `suspended` state soon anyways. But actually it seems simpler to just `close()` the tasks here and now altogether